src/Tool/Commands/ServiceControlCommand.cs

```diff
@@ -122,25 +124,181 @@ async Task<QueueThroughput[]> SampleData(int minutes, CancellationToken cancella
     {
         var monitoringDataUrl = $"{monitoringUrl}/monitored-endpoints?history={minutes}";
 
-        using (var stream = await http.GetStreamAsync(monitoringDataUrl, cancellationToken))
-        using (var reader = new StreamReader(stream))
-        using (var jsonReader = new JsonTextReader(reader))
+        var arr = await GetServiceControlData<JArray>(monitoringDataUrl, cancellationToken);
+
+        var results = arr.Select(token =>
         {
-            var arr = serializer.Deserialize<JArray>(jsonReader);
+            var name = token["name"].Value<string>();
+            var throughputAvgPerSec = token["metrics"]["throughput"]["average"].Value<double>();
+            var throughputTotal = throughputAvgPerSec * minutes * 60;
 
-            return arr.Select(token =>
+            return new QueueThroughput
             {
-                var name = token["name"].Value<string>();
-                var throughputAvgPerSec = token["metrics"]["throughput"]["average"].Value<double>();
-                var throughputTotal = throughputAvgPerSec * minutes * 60;
+                QueueName = name,
+                Throughput = (int)throughputTotal
+            };
+        }).ToList();
+
+        var monitoredNames = results.Select(ep => ep.QueueName).ToHashSet(StringComparer.OrdinalIgnoreCase);
+
+        foreach (var endpoint in knownEndpoints.Where(ep => !monitoredNames.Contains(ep.Name) && ep.AuditedMessages))
+        {
+            var fromAuditing = await GetThroughputFromAudits(endpoint.Name, cancellationToken);
+            if (fromAuditing is not null)
+            {
+                results.Add(fromAuditing);
+            }
+        }
+
+        return results.ToArray();
+    }
+
+    async Task<QueueThroughput> GetThroughputFromAudits(string endpointName, CancellationToken cancellationToken)
+    {
+        Console.WriteLine($"Getting throughput from {endpointName} using audit data.");
+
+        var sampleStart = DateTime.UtcNow.AddMinutes(-minutesPerSample);
+
+        async Task<AuditBatch> GetPage(int page)
+        {
+            Console.WriteLine($"  * Getting page {page}");
+            return await GetAuditBatch(endpointName, page, auditSamplingPageSize, cancellationToken);
```

Comment:
Nit: Maybe another particularism, but its rare for me to use or see const values camelCased.